Raven.Database/Server/WebApi/RequestManager.cs

```diff
@@ -269,7 +271,57 @@ public void Dispose()
 			return response;
 		}
 
-		/// <summary>
+	    private long PreActionLog(IResourceApiController controller)
+	    {
+            LogHttpRequestStatsParams logHttpRequestStatsParam = null;
+            try
+            {
+                StringBuilder sb = null;
+                if (controller.CustomRequestTraceInfo != null && controller.CustomRequestTraceInfo.Count > 0)
+                {
+                    sb = new StringBuilder();
+                    foreach (var action in controller.CustomRequestTraceInfo)
+                    {
+                        action(sb);
+                        sb.AppendLine();
+                    }
+                    while (sb.Length > 0)
+                    {
+                        if (!char.IsWhiteSpace(sb[sb.Length - 1]))
+                            break;
+                        sb.Length--;
+                    }
+                }
+                var innerRequest = controller.InnerRequest;
+                var httpRequestHeaders = innerRequest.Headers;
+                var httpContentHeaders = innerRequest.Content == null ? null : innerRequest.Content.Headers;
+                Stopwatch dummyStopwatch = new Stopwatch();
+                logHttpRequestStatsParam = new LogHttpRequestStatsParams(
```

Comment:
We haven't done anything to the request yet, we just need the URL / database, nothing more.